tests/test_ote_training.py

```diff
@@ -0,0 +1,613 @@
+
+        return argnames, argvalues, ids
+
+    @pytest.fixture(scope='class')
+    def cached_from_prev_test_fx(self):
+        """
+        This fixture is intended for storying the impl class OTETrainingImpl.
+        This object should be persistent between tests while the tests use the same parameters
+        -- see the method _clean_cache_if_parameters_changed below that is used to clean
+        the impl if the parameters are changed.
+        """
+        return dict()
+
+    @staticmethod
+    def _clean_cache_if_parameters_changed(cache, **kwargs):
+        is_ok = True
+        for k, v in kwargs.items():
+            is_ok = is_ok and (cache.get(k) == v)
+        if is_ok:
+            logger.info('TestOTETraining: parameters were not changed -- cache is kept')
+            return
+
+        for k in list(cache.keys()):
+            del cache[k]
+        for k, v in kwargs.items():
+            cache[k] = v
+        logger.info('TestOTETraining: parameters were changed -- cache is cleaned')
+
+    @staticmethod
+    def _update_impl_in_cache(cache,
+                              dataset_name, model_name, num_training_iters,
+                              dataset_definitions, template_paths):
+        # TODO(lbeynens): make this a fixture with conditional ops inside
+        TestOTETraining._clean_cache_if_parameters_changed(cache,
+                                                           dataset_name=dataset_name,
+                                                           model_name=model_name)
+        if 'impl' not in cache:
+            logger.info('TestOTETraining: creating OTETrainingImpl')
+            dataset_params = _get_dataset_params_from_dataset_definitions(dataset_definitions, dataset_name)
+            template_path = _make_path_be_abs(template_paths[model_name], template_paths[ROOT_PATH_KEY])
+            cache['impl'] = OTETrainingImpl(dataset_params, template_path, num_training_iters)
+
+        return cache['impl']
+
+    @pytest.fixture
+    def data_collector_fx(self, request):
+        setup = deepcopy(request.node.callspec.params)
+        setup["environment_name"] = os.environ.get("TT_ENVIRONMENT_NAME", "no-env")
+        setup["test_type"] = os.environ.get("TT_TEST_TYPE", "no-env")
+        setup["scenario"] = "api"
+        setup["test"] = request.node.name
+        setup["subject"] = "custom-object-detection"
+        setup["project"] = "ote"
+        logger.info(f'creating DataCollector: setup=\n{pformat(setup, width=140)}')
+        data_collector = DataCollector(name='TestOTETraining',
+                                       setup=setup)
+        with data_collector:
+            logger.info('data_collector is created')
+            yield data_collector
+        logger.info('data_collector is released')
+
+    @e2e_pytest_performance
+    def test_ote_01_training(self, dataset_name, model_name, num_training_iters,
```

Comment:
What does `01` mean.

Reply:
It is used to order the tests.
I make a trick to store the result of previous test for the next tests if parameters were not changed
(see the method `_update_impl_in_cache` of the class `TestOTETraining`, also see the method `run_ote_training_once` of the class)
Using this trick the tests may be run in any order and the test results (pass/fail) will be the same for any order, but the logs will look more logical if the order is "train => evaluate => export => export after evaluate", so I insert the numbers to order the tests.